src/metabase/pulse/render/body.clj

```diff
@@ -671,66 +634,103 @@
   This can have really high cardinality of series but the JS viz will complain about more than 100 already"
-  [chart-type joined-rows _x-cols _y-cols {:keys [viz-settings] :as data}]
+  [chart-type joined-rows x-cols _y-cols {:keys [viz-settings] :as data} card-name]
   (let [grouped-rows (group-by #(second (first %)) joined-rows)
         groups       (keys grouped-rows)]
     (for [[idx group-key] (map-indexed vector groups)]
       (let [row-group          (get grouped-rows group-key)
             selected-row-group (mapv #(vector (ffirst %) (first (second %))) row-group)
-            card-name          (or (series-setting viz-settings group-key :name)
-                                   (series-setting viz-settings group-key :title)
-                                   group-key)
-            card-color         (or (series-setting viz-settings group-key :color)
-                                   (nth colors idx))
+            column-name          (or (series-setting viz-settings group-key :name)
+                                   (series-setting viz-settings group-key :title))
             card-type          (or (series-setting viz-settings group-key :display)
                                    chart-type
                                    (nth default-combo-chart-types idx))
             y-axis-pos         (or (series-setting viz-settings group-key :axis)
                                    (nth (default-y-pos data axis-group-threshold) idx))]
-        {:name          card-name
-         :color         card-color
+        {:name          column-name
+         :cardName      card-name
          :type          card-type
          :data          selected-row-group
-         :yAxisPosition y-axis-pos}))))
+         :yAxisPosition y-axis-pos
+         :column        (second x-cols)
+         :breakoutValue group-key}))))
+
+(defn- axis-row-fns
+  [card data]
+  [(or (ui-logic/mult-x-axis-rowfn card data) #(vector (first %)))
+   (or (ui-logic/mult-y-axis-rowfn card data) #(vector (second %)))])
+
+(defn- card-result->series
+  "Helper function for `render-multiple-lab-chart` that turns a card query result into a series-settings map in the shape expected by `js-svg/combo chart` (and the combo-chart js code)."
+  [idx result]
+  (let [card            (:card result)
+        data            (get-in result [:result :data])
+        display         (:display card)
+        [x-fn y-fn]     (axis-row-fns card data)
+        enforced-type   (if (= display :scalar) :bar display)
+        card-name       (:name card)
+        viz-settings    (:visualization_settings card)
+        joined-rows     (map (juxt x-fn y-fn)
+                             (common/row-preprocess x-fn y-fn (:rows data)))
+        [x-cols y-cols] ((juxt x-fn y-fn) (get-in result [:result :data :cols]))
+        y-axis-position (nth (default-y-pos data axis-group-threshold) idx)]
```

Comment:
@adam-james-v I modify this `y-axis-position` from your suggestion.

```diff
+y-axis-position (nth (default-y-pos data axis-group-threshold) idx)]
-y-axis-position (take idx (default-y-pos data axis-group-threshold))]
```

I think we want to get a single element rather than a list.

Reply:
Yep, good catch